crates/sui-indexer/src/models_v2/checkpoints.rs

```diff
@@ -0,0 +1,135 @@
+                .iter()
+                .map(|tx| tx.into_inner().to_vec())
+                .collect(),
+            network_total_transactions: c.network_total_transactions as i64,
+            previous_checkpoint_digest: c
+                .previous_checkpoint_digest
+                .as_ref()
+                .map(|d| (*d).into_inner().to_vec()),
+            end_of_epoch: c.end_of_epoch,
+            timestamp_ms: c.timestamp_ms as i64,
+            total_gas_cost: c.total_gas_cost,
+            computation_cost: c.computation_cost as i64,
+            storage_cost: c.storage_cost as i64,
+            storage_rebate: c.storage_rebate as i64,
+            non_refundable_storage_fee: c.non_refundable_storage_fee as i64,
+            checkpoint_commitments: c.checkpoint_commitments.clone(),
+            validator_signature: c.validator_signature.clone(),
+        }
+    }
+}
+
+impl StoredCheckpoint {
+    pub fn into_rpc(
+        self,
+        end_of_epoch_data: Option<EndOfEpochData>,
+    ) -> Result<RpcCheckpoint, IndexerError> {
+        let parsed_digest: CheckpointDigest =
+            bcs::from_bytes(&self.checkpoint_digest).map_err(|e| {
+                IndexerError::SerdeError(format!(
+                    "Failed to decode checkpoint digest: {:?} with err: {:?}",
+                    self.checkpoint_digest, e
+                ))
+            })?;
+
+        let parsed_previous_digest: Option<CheckpointDigest> = self
+            .previous_checkpoint_digest
+            .map(|digest| {
+                bcs::from_bytes(&digest).map_err(|e| {
+                    IndexerError::SerdeError(format!(
+                        "Failed to decode previous checkpoint digest: {:?} with err: {:?}",
+                        digest, e
+                    ))
+                })
+            })
+            .transpose()?;
+        let transactions: Vec<TransactionDigest> = self
+            .tx_digests
+            .into_iter()
+            .map(|tx| {
+                bcs::from_bytes(&tx).map_err(|e| {
+                    IndexerError::SerdeError(format!(
+                        "Failed to decode transaction digest: {:?} with err: {:?}",
+                        tx, e
+                    ))
+                })
+            })
+            .collect::<Result<Vec<TransactionDigest>, IndexerError>>()?;
+        let validator_signature = bcs::from_bytes(&self.validator_signature).map_err(|e| {
+            IndexerError::SerdeError(format!(
+                "Failed to decode validator signature: {:?} with err: {:?}",
+                self.validator_signature, e
+            ))
+        })?;
+
+        let checkpoint_commitments =
+            bcs::from_bytes(&self.checkpoint_commitments).map_err(|e| {
+                IndexerError::SerdeError(format!(
+                    "Failed to decode checkpoint commitments: {:?} with err: {:?}",
+                    self.checkpoint_commitments, e
+                ))
+            })?;
```

Comment:
i think this is just raw bytes too

Reply:
i'm leaving this as bcs because it's [an enum ](https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/messages_checkpoint.rs#L89)